### PR TITLE
TST: adjust other very slow tests

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Followup to gh-20468

#### What does this implement/fix?
This adjust some very slow (>5s) tests, either marking them `xslow` or skipping parts of them.

#### Additional information
Code-owners, LMK if the adustment in your code looks OK.

### CODE REVIEW

File: scipy/special/tests/test_extending.py
Comment (rgommers) [inline scipy/special/tests/test_extending.py:10]: This one is not okay. This is basic test coverage for the `cython_special` module.

File: scipy/interpolate/tests/test_gil.py
Comment (rgommers) [inline scipy/interpolate/tests/test_gil.py:31]: This one seems fine to mark as `xslow` indeed. Kinda pointless to run really slow tests that are marked as `xfail`.

File: scipy/_lib/tests/test_import_cycles.py
Comment (rgommers) [inline scipy/_lib/tests/test_import_cycles.py:11]: This one is important, should be `slow` not `xslow`.

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks for working on this Matt. 

I think to make this work, we should have one CI job running `xslow` tests. And we should add it as a proper test mode (`python dev.py test -m xslow`). Right now `xslow` tests are effectively never run (maybe by a dev a few times per release cycle).

A few of these seem fine to move into `xslow`, but you're also effectively removing test coverage for all our Cython APIs and for import cycles here (putting the former in `'fast'` was a very recent decision in gh-20422) - that cannot be right.

The criterion here is not only "how long does a single test take" but more "is this test worth this much time in every fast/full run".

**Comment by ev-br** [comment]
> we should have one CI job running xslow tests. 

Worth trying, but I wouldn't be surprised if that would require e.g. an xxslow marker to exclude tesrs with very extravagant requirements (memory, cpu etc)

File: scipy/special/tests/test_extending.py
Comment (mdhaber) [reply scipy/special/tests/test_extending.py:10]: I made them all xslow knowing that everyone would be pinged as a code owner, and I would be corrected 
It worked! We'll take it out of this PR and give it more time to run in the fail_slow PR.

Comment (mdhaber) [reply scipy/special/tests/test_extending.py:10]:
```suggestion
```

**Review by mdhaber — COMMENTED** [review]
@rgommers I adjusted the ones you mentioned and the `test_cython` tests in `linalg` and `optimize`. The rest look ok to you? Any I should check with someone else about?

File: scipy/optimize/tests/test_minimize_constrained.py
Comment (mdhaber) [inline scipy/optimize/tests/test_minimize_constrained.py:460]:
```suggestion
                        # BoundedRosenbrock(),  # fails on one CI platform?
```

File: scipy/optimize/tests/test_minimize_constrained.py
Comment (mdhaber) [inline scipy/optimize/tests/test_minimize_constrained.py:460]:
```suggestion
                        BoundedRosenbrock(),
```

File: scipy/integrate/tests/test_quadpack.py
Comment (mdhaber) [inline scipy/integrate/tests/test_quadpack.py:345]: There are plenty of tests for 2D improper integrals, and there is other coverage of 3D integrals. Both of these functions just call `nquad`, so I think it's safe enough to run these only selectively, like when work is done on the functions (rare).

File: scipy/interpolate/tests/test_rgi.py
Comment (mdhaber) [inline scipy/interpolate/tests/test_rgi.py:479]: Plenty of other coverage.

File: scipy/optimize/tests/test__differential_evolution.py
Comment (mdhaber) [inline scipy/optimize/tests/test__differential_evolution.py:1342]: This is just one of many tests. This one happens to be very slow, and it fails on some platforms, so probably no harm in making it xslow.

File: scipy/optimize/tests/test_extending.py
Comment (mdhaber) [inline scipy/optimize/tests/test_extending.py:8]:
```suggestion
```

File: scipy/optimize/tests/test_minimize_constrained.py
Comment (mdhaber) [inline scipy/optimize/tests/test_minimize_constrained.py:216]:
```suggestion
```

File: scipy/optimize/tests/test_minimize_constrained.py
Comment (mdhaber) [inline scipy/optimize/tests/test_minimize_constrained.py:447]: Simply converted the `for` loops to `parametrize`.

File: scipy/optimize/tests/test_minimize_constrained.py
Comment (mdhaber) [inline scipy/optimize/tests/test_minimize_constrained.py:482]: Started getting a failure in just one case on just the Accelerate platform. I thought it was just a matter of the guess `x0` changing, but that doesn't seem to be the case. Not sure what's happening, but it's not too concerning, so I'll just open an issue when this merges.

File: scipy/sparse/linalg/tests/test_propack.py
Comment (mdhaber) [inline scipy/sparse/linalg/tests/test_propack.py:92]: I think I wrote this test, and I don't think there's anything very special about it; it's just one that happened to be distributed with PROPACK. We have other tests that don't take as long.

File: scipy/spatial/tests/test_qhull.py
Comment (mdhaber) [inline scipy/spatial/tests/test_qhull.py:347]: The inclusion of "more" in the name suggests that it's not essential.

File: scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py
Comment (mdhaber) [inline scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py:536]: Another option would be to @parametrize rather than looping and reduce the number of cases.

**Review by mdhaber — COMMENTED** [review]
Made the requested changes. Included some comments here to justify why it's probably ok to run the tests less often.

**Comment by rgommers** [comment]
The current selection (after applying the proposed changes for `test_extending.py`) seem fine to me.

**Review by rgommers — APPROVED** [review]

File: scipy/optimize/tests/test_extending.py
Comment (mdhaber) [inline scipy/optimize/tests/test_extending.py:8]:
```suggestion

@pytest.mark.skipif(platform.machine() in ["wasm32", "wasm64"],
```

File: scipy/spatial/tests/test_qhull.py
Comment (tylerjereddy) [reply scipy/spatial/tests/test_qhull.py:347]: `slow` seems like the right call to me instead of `xslow` here. Checking locally:

`python dev.py test -m full -t scipy/spatial/tests/test_qhull.py::TestUtilities::test_more_barycentric_transforms -- --durations=2`

`0.67s call     build-install/lib/python3.11/site-packages/scipy/spatial/tests/test_qhull.py::TestUtilities::test_more_barycentric_transforms`

That seems reasonable, under a second, and only when `full` is requested.

**Comment by tylerjereddy** [comment]
Let's revert the `spatial` one before merging IMO, as noted above.

File: scipy/spatial/tests/test_qhull.py
Comment (mdhaber) [reply scipy/spatial/tests/test_qhull.py:347]: That's fine. Note that the criterion for inclusion here was that these tests took `>5s` on my machine.

Comment (mdhaber) [inline scipy/spatial/tests/test_qhull.py:347]:
```suggestion
    @pytest.mark.slow
```

**Review by rgommers — COMMENTED** [review]
Thanks Matt. Almost there, but CI shouldn't start failing if we merge this, so I'd like to double check on the `TestTrustRegionConstr` thing.

**Review by mdhaber — COMMENTED** [review]
Responded to comments; all tests pass.

File: scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py
Comment (rgommers) [reply scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py:636]: This breaks a hard requirement that we have: tests should be reproducible. So this change cannot be right. Why not mark some subset as slow (or even xslow). Or, if achieving good test coverage in a reasonable amount of time really is that problematic, then this may be a fit for `hypothesis`. We default to fixed seeds, but it's possible with `hypothesis` to randomly sample the subspace in a more controlled way (plus if it fails, it tells you the reproducer).

Comment (mdhaber) [reply scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py:636]: This has a fixed seed.

**Review by rgommers — APPROVED** [review]
Okay, let's give this a go, it's been reviewed in enough detail. Thanks Matt!

**Pull request merged**
